sql/plan/create_view.go

```diff
@@ -0,0 +1,112 @@
+package plan
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/src-d/go-mysql-server/sql"
+)
+
+// CreateView is a node representing the creation (or replacement) of a view,
+// which is defined by the Child node. The Columns member represent the
+// explicit columns specified by the query, if any.
+type CreateView struct {
+	UnaryNode
+	database  sql.Database
+	Name      string
+	Columns   []string
+	Catalog   *sql.Catalog
+	IsReplace bool
+}
+
+// NewCreateView creates a CreateView node with the specified parameters,
+// setting its catalog to nil.
+func NewCreateView(
+	database sql.Database,
+	name string,
+	columns []string,
+	definition *SubqueryAlias,
+	isReplace bool,
+) *CreateView {
+	return &CreateView{
+		UnaryNode{Child: definition},
+		database,
+		name,
+		columns,
+		nil,
+		isReplace,
+	}
+}
+
+// View returns the view that will be created by this node.
+func (create *CreateView) View() sql.View {
+	return sql.NewView(create.Name, create.Child)
+}
+
+// Children implements the Node interface. It returns the Child of the
+// CreateView node; i.e., the definition of the view that will be created.
+func (create *CreateView) Children() []sql.Node {
+	return []sql.Node{create.Child}
+}
+
+// Resolved implements the Node interface. This node is resolved if and only if
+// the database and the Child are both resolved.
+func (create *CreateView) Resolved() bool {
+	_, ok := create.database.(sql.UnresolvedDatabase)
+	return !ok && create.Child.Resolved()
+}
+
+// RowIter implements the Node interface. When executed, this function creates
+// (or replaces) the view. It can error if the CraeteView's IsReplace member is
+// set to false and the view already exists. The RowIter returned is always
+// empty.
+func (create *CreateView) RowIter(ctx *sql.Context) (sql.RowIter, error) {
+	view := sql.NewView(create.Name, create.Child)
+
+	if create.IsReplace {
+		_ = create.Catalog.ViewRegistry.Delete(create.database.Name(), view.Name())
```

Comment:
why is this error ignored? should we not fail in this case?

Reply:
If the view is not there, we will just create it. It just means that the user used the OR REPLACE keywords but the view was not actually there.

Reply:
Right now this is ok because it's in memory, but when persistence is implemented there will be more types of errors that can be returned by delete, which may be error-prone in the future and we may forget handling the error here.

I think it would be best to do the following:

```go
err := create.Catalog.ViewRegistry.Delete(create.database.Name(), view.Name())
if err != nil && !ErrViewNotExists.Is(err) {
    return err
}
```

In this case we will ignore only the specific error we want to ignore and handle any other possible error that may be returned.

Reply:
Done in https://github.com/src-d/go-mysql-server/pull/858/commits/ccdbd1758487f9b56d61d92b8849aa799ba027ed